packages/autocomplete-types/index.d.ts

```diff
@@ -175,15 +175,22 @@ declare namespace Fig {
    */
   type ExecuteShellCommandFunction = (commandToExecute: string) => Promise<string>;
 
+  type CacheStrategy = "stale-while-revalidate" | "max-age";
+
   type Cache = {
+    /**
+     * The Cache strategy to use for this generator.
+     * @defaultValue 'max-age'
+     */
+    strategy?: CacheStrategy;
     /**
      * The time to live for the cache in milliseconds.
      * @example
      * 3600
      */
     ttl: number;
```

Comment:
Does `ttl` still need to be specified with `strategy: "stale-while-revalidate"`?

Reply:
It doesn't but it is required for max-age. The optimal solution would be to make strategy an object that only contains a mandatory ttl for 'max-age' but that would be breaking. Since we are defaulting on 'max-age' strategy I would keep ttl as required. To disable it for swr we can still set it to 0.

Reply:
Would something like this be fine? I'd be totally okay with the `ttl: 0` workaround too 🙂
```
namespace Fig {
  export type CacheTimeToLive = {
    stategy?: "max-age";
    ttl: number;
    cacheByDirectory?: boolean
  }
  export type CacheStaleWhileRevalidate = {
    strategy: "stale-while-revlidate";
    cacheByDirectory?: boolean;
  }
  export type Cache =
    | CacheTimeToLive
    | CacheStaleWhileRevalidate;
}

const ttl: Fig.Cache = { ttl: 30000 };
const swr: Fig.Cache = { strategy: "stale-while-revlidate" };
```

Reply:
Definitely better!